python/src/aiconfig/editor/client/src/components/prompt/model_settings/ModelSettingsRenderer.tsx

```diff
@@ -93,6 +97,8 @@ export default memo(function ModelSettingsRenderer({
         >
           {rawJSONToggleButton}
           <ModelSettingsSchemaRenderer
+            // Use the model name as the key to force re-mount / updated state when model changes
```

Comment:
Could this cause an issue if we have multiple props where we re-use the default model and hence have the same key for two separate `ModelSettingsSchemaRenderer` objects? Or is React smart enough to understand the hierarchy and know that even though we set the same keys, they're not the same objects?

Reply:
> multiple props where we re-use the default model

Sorry typo, meant to say "multiple prompts"